### ISSUE TITLE
Jobs Take Increasingly Longer to Start as More Data is Added to the Repo

### ISSUE THREAD

**Issue by dgeorg42** [pachyderm/pachyderm#6577]
Running on Beta 2, I have noticed that as the amount of data in a repo increases, the time it takes for a job to actually start also increases (and pretty significantly). The number of commits in the repo _may_ play a role as well; I haven't controlled for that in my testing.

I have a pipeline with a glob pattern of `/` so every file in the root of the repo is taken together as a single datum. During my testing, I have uploaded (via `pachctl put file`) 20,000 files at a time until the repo contains 200,000 files (thus, 10 commits). After each upload was completed, `pachctl list job` would show something like the following: 
```
ID                               PIPELINE     STARTED DURATION RESTART PROGRESS  DL UL STATE 
85f3bd85f508411bbf451d7350d7920c benchmarking -       -        0       0 + 0 / 0 0B 0B -     
```
The job would sit like that for a while before eventually transitioning into the `running` state and displaying something other than "-" in the STARTED column. Each job took longer than the previous one to transition.

Once my test set had completed, I ran `pachctl update pipeline -f benchmarking.json --reprocess` specifically to capture logs for this ticket.  I observed the job sitting in this limbo-like hyphenated state for ~7 minutes. During this time, I was tailing some of the logs, and in the pipeline's storage log (attached) I noticed one particular call to InspectCommit which took about 5 and a half minutes to complete. Once the job finally transitioned into the "running" state, it only took about 4 minutes to run.

[storage.log](https://github.com/pachyderm/pachyderm/files/6852733/storage.log)

**Comment by brycemcanally** [comment]
Are the file paths randomly generated in this workload? I suspect the time is being spent compacting (you are probably seeing the time through inspect commit because it is the first operation that will require compaction to get the size). We compact pretty aggressively right now which is not good for a workload that adds files evenly distributed across the key space because it can involve rewriting a lot of data. We have a fixed delay configuration that we will probably want to bump up higher and expose so it can be tuned to the workload.

**Comment by dgeorg42** [comment]
@brycemcanally - No, all of the files exist at the root of the repo.

**Comment by brycemcanally** [comment]
Yeah, but how are the actual file paths for the files in the root directory generated? The order in which the files get into the root directory matters in terms of what object storage chunks will get created during compaction.

**Comment by dgeorg42** [comment]
I'm not sure I completely understand what you're asking.  So on my local disk I have a directory called "runtests-temp", into which I copy whatever files I want to upload into the repo. For these particular tests, I start with a series of image files named 000001.jpg all the way up to 020000.jpg (the first 20,000 files) and I load them to the repo by doing `pachctl put file -r bench@master:/ -f ./runtests-temp`. After the pipeline kicks off and runs, my script empties the `runtests-temp` directory, and copies in the next batch of files to be loaded (in this case 020001.jpg through 040000.jpg).  This continues until there's 200,000 files in the repo, all sitting at the root level - so there's no path per se, unless you count "/" as a path.
